include/libhal-nunchuck/nunchuck.hpp

```diff
@@ -14,7 +14,83 @@
 
 #pragma once
 
+#include <cstdint>
+
+#include <libhal/i2c.hpp>
+
 namespace hal::nunchuck {
-struct nunchuck_replace_me
-{};
+/**
+ * @brief Driver for nunchuck i2c gamepad
+ *
+ */
+class nunchuck
+{
+public:
+  /**
+   * @brief Holds information about gamepad state, and contains functions to
+   * return translated data.
+   *
+   */
+  struct gamepad_data
+  {
+    /// @brief byte array containing data about gamepad state.
+    std::array<hal::byte, 6> data;
+    /**
+     * @brief Get x-axis data of the joystick.
+     * @return uint8_t representing the x-axis data of the joystick. Center
+     * position is 128. 0-127 indicates the stick is pointed left and 129-255
+     * indicates the stick is pointed right.
+     */
+    std::uint8_t joystick_x();
+    /**
+     * @brief Get y-axis data of the joystick.
+     * @return uint8_t representing the y-axis data of the joystick. Center
+     * position is 128. 0-127 indicates the stick is pointed dwon and 129-255
+     * indicates the stick is pointed up.
+     */
+    std::uint8_t joystick_y();
+    /**
+     * @brief Get x-axis data of the accelerometer sensor.
+     * @return uint16_t representing the x-axis data of the accelerometer.
+     * Center position is about 512. Reducing values means left movement,
+     * increasing values means right movement.
+     */
+    std::uint16_t accelerometer_x();
+    /**
+     * @brief Get y-axis data of the accelerometer sensor.
+     * @return uint16_t representing the y-axis data of the accelerometer.
+     * Center position is about 512. Reducing values means backward movement,
+     * increasing values means forward movement.
+     */
+    std::uint16_t accelerometer_y();
+    /**
+     * @brief Get z-axis data of the accelerometer sensor.
+     * @return uint16_t representing the z-axis data of the accelerometer.
+     * Center position is about 512. Reducing values means downward movement,
+     * increasing values means upward movement.
+     */
+    std::uint16_t accelerometer_z();
+    /**
+     * @brief Get z button pressed state.
+     * @return Boolean representing if the z button is pressed or not.
+     */
+    bool z_button();
+    /**
+     * @brief Get c button pressed state.
+     * @return Boolean representing if the c button is pressed or not.
+     */
+    bool c_button();
+  };
+
+  /**
+   * @brief Create a nunchuck object
+   * @param p_i2c - i2c object that the nunchuck will use
+   */
+  nunchuck(hal::i2c& p_i2c);
+
+private:
+  gamepad_data driver_read();
```

Comment:
```suggestion

  gamepad_data read();
private:
```
Should make this public so the user can call it to get the gamepad data. We can also reduce the name to just `read`. `driver_read` would be used if this was implementing an interface.

Reply:
done